src/Chainweb/Pact/Backend/RelationalCheckpointer.hs

```diff
@@ -155,11 +155,20 @@ doSave dbenv hash = runBlockEnv dbenv $ do
         -> BlockHandler SQLiteEnv ()
     createNewTables bh = mapM_ (\tn -> createUserTable (Utf8 tn) bh)
 
+-- | Discards all transactions since the most recent @Block@ savepoint and
+-- removes the savepoint from the transaction stack.
+--
 doDiscard :: Db -> IO ()
 doDiscard dbenv = runBlockEnv dbenv $ do
     clearPendingTxState
     rollbackSavepoint Block
 
+    -- @ROLLBACK TO n@ only rolls back updates up to @n@ but doesn't remove the
+    -- savepoint. In order to also pop the savepoint from the stack we commit it
+    -- (as empty transaction). <https://www.sqlite.org/lang_savepoint.html>
+    --
+    commitSavepoint Block
```

Comment:
My understanding of the code was that `rollBackSavepoint` and `commitSavepoint` closely resemble behavior of the respective sqlite commands and the `do*` commands at our own transaction DSL on top of that.

It may confuse people if we use sqlite terminology with different semantics. Not sure, though.

Reply:
Ok! that's a great argument for leaving it as is.